script/playercomponent.js

```diff
@@ -228,37 +250,41 @@ define(
       function attemptCdnFailover (errorProperties, bufferingTimeoutError) {
         var shouldFailover = MediaResilience.shouldFailover(mediaMetaData.urls.length, getDuration(), getCurrentTime(), getLiveSupport(device), windowType, transferFormat);
 
+        var failover = function (time) {
+          cdnFailover(time, thenPause, errorProperties, bufferingTimeoutError);
+        };
+        var errorCallback = function () {
+          bubbleFatalError(errorProperties, bufferingTimeoutError);
+        };
+
         if (shouldFailover) {
           var thenPause = playbackStrategy.isPaused();
           tearDownMediaElement();
 
           var failoverTime = getCurrentTime();
-          if (transferFormat === TransferFormats.HLS && LiveSupportUtils.needToGetManifest(windowType, getLiveSupport(device))) {
-            manifestReloadFailover(failoverTime, thenPause, errorProperties, bufferingTimeoutError);
-          } else {
-            cdnFailover(failoverTime, thenPause, errorProperties, bufferingTimeoutError);
-          }
+          reloadManifest(failoverTime, failover, errorCallback);
         } else {
-          bubbleFatalError(errorProperties, bufferingTimeoutError);
+          errorCallback();
         }
       }
 
-      function manifestReloadFailover (failoverTime, thenPause, errorProperties, bufferingTimeoutError) {
-        ManifestLoader.load(
-          bigscreenPlayerData.media.urls,
-          bigscreenPlayerData.serverDate,
-          {
-            onSuccess: function (manifestData) {
-              var windowOffset = manifestData.time.windowStartTime - getWindowStartTime();
-              bigscreenPlayerData.time = manifestData.time;
-              failoverTime -= windowOffset / 1000;
-              cdnFailover(failoverTime, thenPause, errorProperties, bufferingTimeoutError);
-            },
-            onError: function () {
-              bubbleFatalError(errorProperties, bufferingTimeoutError);
+      function reloadManifest (time, successCallback, errorCallback) {
```

Comment:
Can we either rename this, or move the `if` logic out to usages of it? It's a bit misleading to call it `reloadManifest` when it might not.

Reply:
I don't think you should care if you need to reload the manifest or not so I'd prefer the if statement to be kept inside.